src/inmanta/server/validate_filter.py

```diff
@@ -203,6 +203,42 @@ def to_query_type(self) -> Optional[Tuple[QueryType, object]]:
         return None
 
 
+class CombinedContainsFilterResourceState(Filter):
+    """Represents a valid ReleasedResourceState constraint,
+    which handles the filters as contains or not contains filters based on their values"""
+
+    # Pydantic doesn't support Generic models on python 3.6
+    field: Optional[Dict[QueryType, List[ReleasedResourceState]]]
+
+    @validator("field", pre=True)
+    @classmethod
+    def parse_field(cls, v: object) -> Optional[Dict[QueryType, ReleasedResourceState]]:
+        if v is None:
+            return None
+        if isinstance(v, list):
+            status_contains_filters = [status_filter for status_filter in v if not status_filter.startswith("!")]
+            status_not_contains_filters = [status_filter[1:] for status_filter in v if status_filter.startswith("!")]
```

Comment:
This is part of the input validation logic. What happens if we pass in something that is not indexable (an integer e.g.)?  or has no startswith method?